Source/CombatExtended/CombatExtended/ExplosionCE.cs

```diff
@@ -198,24 +218,62 @@ public virtual IEnumerable<IntVec3> ExplosionCellsToHit
                         //Added code
                         if (aboveRoofs)
                         {
+                            Log.Message("isAbove");
                             if ((!roofed && GenSight.LineOfSight(Position, intVec, Map, false, null, 0, 0))
                                || !intVec.Roofed(Map))
                             {
                                 openCells.Add(intVec);
                             }
                         }
-                        else if (GenSight.LineOfSight(Position, intVec, Map, true, null, 0, 0))
+                        else 
                         {
-                            if (needLOSToCell1.HasValue || needLOSToCell2.HasValue)
-                            {
-                                bool flag = needLOSToCell1.HasValue && GenSight.LineOfSight(needLOSToCell1.Value, intVec, Map, false, null, 0, 0);
-                                bool flag2 = needLOSToCell2.HasValue && GenSight.LineOfSight(needLOSToCell2.Value, intVec, Map, false, null, 0, 0);
-                                if (!flag && !flag2)
-                                {
-                                    continue;
-                                }
+                            if (GenSight.LineOfSight(Position, intVec, Map, true, null, 0, 0))
+                            {
+                                if (needLOSToCell1.HasValue || needLOSToCell2.HasValue)
+                                {
+                                    bool flag = needLOSToCell1.HasValue && GenSight.LineOfSight(needLOSToCell1.Value, intVec, Map, false, null, 0, 0);
+                                    bool flag2 = needLOSToCell2.HasValue && GenSight.LineOfSight(needLOSToCell2.Value, intVec, Map, false, null, 0, 0);
+                                    if (!flag && !flag2)
+                                    {
+                                        continue;
+                                    }
+                                }
+                                openCells.Add(intVec);
+                            }
+                            else
+                            {
+                                var sightPoints = GenSight.PointsOnLineOfSight(Position, intVec);
+
+                                foreach (var pos in sightPoints)
+                                {
+                                    Log.Message(pos.ToString().Colorize(Color.green));
+                                }
+
+                                var barrier = barriers.Find(x => sightPoints.Contains(x.Position));
+
+                                Log.Message("1");
+                                if (barrier != null && barrier.Label != null)

```

Comment:
Why does the barrier need a label?

Reply:
completely forgot to remove debugging parts and redundant checks which were made for testing